src/key.rs

```diff
@@ -292,6 +292,31 @@ impl SecretKey {
     pub fn sign_ecdsa(&self, msg: Message) -> ecdsa::Signature {
         SECP256K1.sign_ecdsa(&msg, self)
     }
+
+    /// Returns the [`KeyPair`] for this [`SecretKey`].
+    ///
+    /// This is equivalent to using [`KeyPair::from_secret_key`].
+    #[inline]
+    pub fn keypair<C: Signing>(&self, secp: &Secp256k1<C>) -> KeyPair {
```

Comment:
If we use `KeyPair`, according to rust naming guidelines this should be `key_pair`, like in `PublicKey/public_key`

Reply:
Double lol, I noticed this while writing this PR too and hoped to no one else would notice because `KeyPair`/keypair has to be addressed in `rust-secp256k1` as well, its an invasive change.

Reply:
I think the problem is that capitalization adds less visual noise than underscoring. Thus, the same API guidelines to add capital letters and underscores in practice do not work well for human eyes.

Personally I do not care if we will use `KeyPair` and `keypair` at the same time - we already doing that all around the library.